include/json/value.h

```diff
@@ -895,9 +902,13 @@ class JSON_API ValueIterator : public ValueIteratorBase {
     return *this;
   }
 
-  reference operator*() const { return deref(); }
-
-  pointer operator->() const { return &deref(); }
+  /*! The return value of non-const iterators can be
+   *  changed, so the these functions are not const
+   *  because the returned references/pointers can be used
+   *  to change state of the base class.
+   */
+  reference operator*() { return deref(); }
```

Comment:
Technically this broke binary compatibility, so this should have been a major version bump.

I guess we get away with this because probably nobody uses this method. If nobody complains, we can keep it.